yggdrasilffi/src/lib.rs

```diff
@@ -0,0 +1,210 @@
+use std::{
+    ffi::{c_char, CStr, CString, NulError},
+    str::Utf8Error,
+};
+
+use unleash_types::{client_features::ClientFeatures, client_metrics::MetricBucket};
+use unleash_yggdrasil::{Context, EngineState};
+
+#[derive(Debug)]
+enum FFIError {
+    Utf8Error,
+    NulError,
```

Comment:
Just a curiosity question: why is this `NulError` as opposed to `NullError` (or maybe `NilError`)? Does it map to something?